File: OOP/1_intro/Reckless+Driver+-+Java/Reckless Driver/src/com/poash/rd/PlayerHealth.java
package com.poash.rd;

public class PlayerHealth {
	private int healthUnits = 100 ;
	private PlayerState playerState = PlayerState.HEALTHY ;
	private void UpdateState() {
		if(healthUnits > 70) {
			playerState = PlayerState.HEALTHY;
		}else if(healthUnits >= 25 && healthUnits <= 70) {
			playerState = PlayerState.INJURED ;
		}else if(healthUnits >=1 && healthUnits < 25) {
			playerState = PlayerState.CRITICAL ;
		}else {
			playerState = PlayerState.DEAD ;
		}
		System.out.println("Current state:" + playerState);
	}
	public int GetUnits() {
		return healthUnits ;
	}
	public void SetUnits(int units) {
		healthUnits = units < 0 ? 0 : units ;
		UpdateState();
	}
	public PlayerState GetPlayerState() {
		return playerState ;
	}
}

File: OOP/1_intro/Reckless+Driver+-+C++/RecklessDriver/RecklessDriver/PlayerHealth.cpp
#include "PlayerHealth.h"


#include <iostream>
#include <string>


void PlayerHealth::UpdateState() {
	if(m_HealthUnits > 70) {
		m_State = PlayerState::HEALTHY ;
	}else if(m_HealthUnits >= 25 && m_HealthUnits <= 70) {
		m_State = PlayerState::INJURED ;
	}else if(m_HealthUnits >= 1 && m_HealthUnits < 25) {
		m_State = PlayerState::CRITICAL ;
	}else {
		m_State = PlayerState::DEAD ;
	}
	std::string stateName{} ;
	switch (m_State) {
	case PlayerState::HEALTHY:
		stateName = "HEALTHY" ;
		break ;
	case PlayerState::INJURED:
		stateName = "INJURED" ;
		break ;
	case PlayerState::CRITICAL:
		stateName = "CRITICAL" ;
		break ;
	case PlayerState::DEAD:
		stateName = "DEAD" ;
		break ;
	}
	std::cout << "Current State :" << stateName << '\n' ;
}

int PlayerHealth::GetUnits() const {
	return m_HealthUnits ;
}

void PlayerHealth::SetUnits(int units) {
	m_HealthUnits = units < 0 ? 0 : units ;
	UpdateState() ;
}

PlayerState PlayerHealth::GetPlayerState() const {
	return m_State ;
}
